x-pack/platform/plugins/shared/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout.tsx

```diff
@@ -13,7 +14,17 @@ export const getShareToSpaceFlyoutComponent = async (): Promise<
   React.FC<ShareToSpaceFlyoutProps>
 > => {
   const { ShareToSpaceFlyoutInternal } = await import('./share_to_space_flyout_internal');
-  return (props: ShareToSpaceFlyoutProps) => {
-    return <ShareToSpaceFlyoutInternal {...props} />;
-  };
+  return React.memo(
+    (props: ShareToSpaceFlyoutProps) => {
+      return <ShareToSpaceFlyoutInternal {...props} />;
+    },
+    (prevProps, nextProps) => {
+      return (
+        prevProps.behaviorContext === nextProps.behaviorContext &&
+        prevProps.savedObjectTarget.id === nextProps.savedObjectTarget.id &&
+        prevProps.savedObjectTarget.type === nextProps.savedObjectTarget.type &&
+        isEqual(prevProps.savedObjectTarget.namespaces, nextProps.savedObjectTarget.namespaces)
+      );
+    }
+  );
```

Comment:
@qn895 I am curious what kind of rendering issues were you seeing. Was this also present before other changes were made in this PR? I'd like to get a better understanding of why this was needed.